http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/websocket/WebSocketTestCases.java

```diff
@@ -0,0 +1,153 @@
+ *  under the License.
+ *
+ */
+
+package org.wso2.carbon.transport.http.netty.websocket;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.messaging.exceptions.ServerConnectorException;
+import org.wso2.carbon.transport.http.netty.config.TransportsConfiguration;
+import org.wso2.carbon.transport.http.netty.config.YAMLTransportConfigurationBuilder;
+import org.wso2.carbon.transport.http.netty.listener.HTTPServerConnector;
+import org.wso2.carbon.transport.http.netty.util.TestUtil;
+import org.wso2.carbon.transport.http.netty.util.clients.websocket.WebSocketClient;
+import org.wso2.carbon.transport.http.netty.util.clients.websocket.WebSocketTestConstants;
+
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import javax.net.ssl.SSLException;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertTrue;
+
+/**
+ * Test class for WebSocket Upgrade
+ */
+public class WebSocketTestCases {
+
+    Logger logger = LoggerFactory.getLogger(WebSocketTestCases.class);
+    private List<HTTPServerConnector> serverConnectors;
+    private WebSocketClient primaryClient = new WebSocketClient();
+    private WebSocketClient secondaryClient = new WebSocketClient();
+
+    @BeforeClass
+    public void setup() {
+        logger.info("\n-------WebSocket Test Cases-------");
+        TransportsConfiguration configuration = YAMLTransportConfigurationBuilder
+                .build("src/test/resources/simple-test-config/netty-transports.yml");
+        serverConnectors = TestUtil.startConnectors(configuration, new WebSocketMessageProcessor());
+    }
+
+    @Test
+    public void handshakeTest() throws URISyntaxException, SSLException {
+        try {
+            assertTrue(primaryClient.handhshake());
+            logger.info("Handshake test completed.");
+        } catch (InterruptedException e) {
+            logger.error("Handshake interruption.");
+            assertTrue(false);
+        }
+    }
+
+    @Test
+    public void testText() throws URISyntaxException, InterruptedException, SSLException {
+        primaryClient.handhshake();
+        String textSent = "test";
+        primaryClient.sendText(textSent);
+        Thread.sleep(3000);
+        String textReceived = primaryClient.getTextReceived();
+        assertEquals("Not received the same text.", textReceived, textSent);
+        logger.info("pushing and receiving text data from server completed.");
+        primaryClient.shutDown();
+    }
+
+    @Test
+    public void testBinary() throws InterruptedException, URISyntaxException, SSLException {
+        primaryClient.handhshake();
+        byte[] bytes = {1, 2, 3, 4, 5};
+        ByteBuffer bufferSent = ByteBuffer.wrap(bytes);
+        primaryClient.sendBinary(bufferSent);
+        Thread.sleep(3000);
```

Comment:
Message delivery is asynchronous. This check whether the sent message has been pushed back from the server. So there is a small time given to ensure the message is received to the client handler.